pkg/resourcemanager/keyvaults/keyvault_manager.go

```diff
@@ -13,8 +13,6 @@ import (
 	"github.com/Azure/azure-service-operator/pkg/resourcemanager"
 )
 
-var AzureKeyVaultManager KeyVaultManager = &azureKeyVaultManager{}
```

Comment:
Why remove this?

Reply:
Instead just do `var _ KeyVaultManager = &azureKeyVaultManager{}` seems reasonable?

Reply:
This was being used by code in other packages - I needed to remove the global variable because it needs to have creds when it's constructed. I'll reinstate the interface assertion part though!